rust/cadeau/src/xmf/recorder.rs

```diff
@@ -139,10 +139,10 @@ impl Recorder {
     ) -> Result<(), RecorderError> {
         // INVARIANT: surface_step * height <= buffer.len()
         // INVARIANT: width * height * 4 <= buffer.len()
-        if !(surface_step * height <= buffer.len()) {
+        if surface_step * height > buffer.len() {
             return Err(RecorderError::BadArgument { name: "buffer" });
         }
-        if !(width * height * 4 <= buffer.len()) {
+        if width * height * 4 > buffer.len() {
```

Comment:
This got reverted again. What’s happening?

Reply:
Oh, I know why, it is because of clippy!

Reply:
Indeed! There is an issue open about exactly that for this lint: https://github.com/rust-lang/rust-clippy/issues/13674
In the meantime, can you annotate the function with this?
```rust
#[expect(clippy::nonminimal_bool)]
```

Reply:
looks like expect is an experiment feature, can I use allow instead?

Reply:
I have to update rust-toolchain.toml